tests/rptest/services/kafka_simple_producer.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2021 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+import re
+import threading
+from ducktape.services.background_thread import BackgroundThreadService
+from rptest.clients.kafka_cli_tools import KafkaCliTools
+from ducktape.utils.util import wait_until
+from typing import Optional
+
+
+class KafkaSimpleProducer(BackgroundThreadService):
+    def __init__(self,
+                 context,
+                 redpanda,
+                 topic,
+                 runtime: int,
+                 compression: Optional[str] = None):
+        super(KafkaSimpleProducer, self).__init__(context, num_nodes=1)
+
+        self._redpanda = redpanda
+        self._topic = topic
+        self._runtime = runtime
+        self._compression = compression
+        self._stopping = threading.Event()
+        self.record_count = 0
+        self.done = False
+
+    def _worker(self, _, node):
+        self._stopping.clear()
+        try:
+            find_class_path = '$(mvn -q exec:exec -Dexec.executable=echo -Dexec.args="%classpath"):target/simple-producer-1.0-SNAPSHOT.jar'
+            cmd = f'cd /opt/simple-producer && java -cp {find_class_path} com.redpanda.simpleproducer.App'
+            cmd += f' {self._redpanda.brokers()} {self._topic} {self._runtime}'
+
+            if self._compression is not None:
+                cmd += f' {self._compression}'
+
+            for line in node.account.ssh_capture(cmd):
```

Comment:
We have a problem there, I've just realized that `ssh_capture` doesn't give us real-time guarantees. Imagine that the app produces log twice faster than `KafkaSimpleProducer` reads it. In this case we may have the following situation:

 - app works for 10 seconds and then crashes
 - KafkaSimpleProducer is still processing the output 15 seconds later 
 - test checks `record_count` and thinks that the app is still running which it was already dead for 5 seconds

Reply:
What you say makes sense, however, that's a problem with all ducktape tests.
IMO, the best way to go forward is with manual testing then. Or at least not use ducktape.

Reply:
OK. So I will re-throw the exception in simple producer app.

Reply:
https://github.com/NyaliaLui/simple-producer/blob/main/src/main/java/com/redpanda/simpleproducer/App.java